cns/restserver/ipam.go

```diff
@@ -661,8 +675,15 @@ func (service *HTTPRestService) GetExistingIPConfig(podInfo cns.PodInfo) ([]cns.
 
 // Assigns a pod with all IPs desired
 func (service *HTTPRestService) AssignDesiredIPConfigs(podInfo cns.PodInfo, desiredIPAddresses []string) ([]cns.PodIpInfo, error) {
+	// Sets the number of desired IPs equal to the number of NCs so that we can get one IP per NC
 	numDesiredIPAddresses := len(desiredIPAddresses)
+	// Creates a slice of PodIpInfo with the size as number of NCs to hold the result for assigned IP configs
 	podIPInfo := make([]cns.PodIpInfo, numDesiredIPAddresses)
+	// if there are no NCs on the NNC there will be no IPs in the pool so return error
+	if numDesiredIPAddresses == 0 {
```

Comment:
Where you basically want to map over `desiredIPAddresses`, it's usually more idiomatic to let the caller pass in a `[]cns.PodIpInfo` that is partially complete (with just the IP addresses). This lets the caller control the allocation, rather than having these intermediate allocations in here.

That also forces the caller of this method to handle the request rather than spreading that responsibility among many other funcs.

If you don't pursue that, this conditional should be at the beginning of the method (so we don't allocate a slice).

Reply:
The majority of these lines were already present, so I don't think a major refactor is needed here, but agree with moving the input validation to be the first thing in the method.

Reply:
I should have emphasized the first part as non-blocking, but moving the conditional as blocking feedback. I still like to provide feedback on the code overall in case someone wants to refactor it... It's unlikely we will ever schedule refactoring this, so PRs provide the only opportunity.